hitbox-tarantool/src/backend.rs

```diff
@@ -0,0 +1,179 @@
+use std::io::{Error, ErrorKind};
+
+use async_trait::async_trait;
+use derive_builder::Builder;
+use hitbox_backend::{
+    serializer::{JsonSerializer, Serializer},
+    BackendError, BackendResult, CacheBackend, CacheableResponse, CachedValue, DeleteStatus,
+};
+use rusty_tarantool::tarantool::{Client, ClientConfig, ExecWithParamaters};
+use serde::{Deserialize, Serialize};
+
+/// Tarantool cache backend based on rusty_tarantool crate.
+///
+/// # Examples
+/// ```
+/// use hitbox_tarantool::TarantoolBackend;
+///
+/// #[tokio::main]
+/// async fn main() {
+///     let mut backend = TarantoolBackendBuilder::default()
+///         .build()
+///         .unwrap();
+///     backend.init().await.unwrap();
+/// }
+/// ```
+#[derive(Clone, Builder)]
+pub struct TarantoolBackend {
+    #[builder(default = "\"hitbox\".to_string()")]
+    user: String,
+    #[builder(default = "\"hitbox\".to_string()")]
+    password: String,
+    #[builder(default = "\"127.0.0.1\".to_string()")]
+    host: String,
+    #[builder(default = "\"3301\".to_string()")]
+    port: String,
+    #[builder(setter(skip))]
+    client: Option<Client>,
+}
+
+impl TarantoolBackend {
+    fn client(&self) -> BackendResult<Client> {
+        let err = Error::new(ErrorKind::Other, "Backend is not initialized");
+        self.client
+            .clone()
+            .ok_or(BackendError::InternalError(Box::new(err)))
+    }
+
+    /// Init backend and configure tarantool instance
+    /// This function is idempotent
+    pub async fn init(&mut self) -> BackendResult<()> {
+        if self.client.is_none() {
+            let client = ClientConfig::new(
+                format!("{}:{}", self.host, self.port),
+                self.user.clone(),
+                self.password.clone(),
+            )
+            .build();
+            self.client = Some(client);
+        }
+
+        self.client()?
+            .eval(
+                "
+                    local space_name = ...
```

Comment:
I think it would be better to move this Lua code into a separate Lua file and load the code from this file when the backend starts. This gives us the ability to update the Lua code without rebuilding the application. This Lua file should be a part of Tarantool backend configuration

Reply:
done